src/hotspot/os/windows/os_windows.cpp

```diff
@@ -4108,6 +4108,10 @@ void os::win32::initialize_windows_version() {
     return;
   }
   strncat(kernel32_path, "\\kernel32.dll", MAX_PATH - ret);
+  // On Windows Nanoserver the kernel32.dll is located in the forwarders subdirectory
+  if (!os::file_exists(kernel32_path)) {
```

Comment:
Should this detection logic be extracted for other scenarios wher we need to special case nano?

Reply:
I've found a registry key that we can reliably use to detect Windows Nanoserver